src/modules/endpoint/sagas/watchFetchCategory.js

```diff
@@ -30,32 +31,45 @@ function * fetchCategory (dataContainer: DataContainer, action: FetchCategoryAct
     const peeking = yield call(isPeeking, city)
 
     const loadCriterion = new ContentLoadCriterion(criterion, peeking)
-    const cityContentLoaded = yield call(loadCityContent, dataContainer, city, language, loadCriterion)
+    const languageValid = yield call(loadCityContent, dataContainer, city, language, loadCriterion)
 
-    if (cityContentLoaded) {
-      // Only proceed if the content is ready to be pushed to the state. If not then the UI automatically displays an
-      // appropriate error
+    // Only get languages if we've loaded them, otherwise we're peeking
+    const cityLanguages = loadCriterion.shouldLoadLanguages() ? yield call(dataContainer.getLanguages, city) : []
+
+    if (languageValid) {
       const [categoriesMap, resourceCache] = yield all([
         call(dataContainer.getCategoriesMap, city, language),
         call(dataContainer.getResourceCache, city, language)
       ])
 
-      // Only get languages if we've loaded them, otherwise we're peeking
-      const cityLanguages = loadCriterion.shouldLoadLanguages() ? yield call(dataContainer.getLanguages, city) : []
-
       const push: PushCategoryActionType = {
         type: `PUSH_CATEGORY`,
         params: { categoriesMap, resourceCache, path, cityLanguages, depth, key, city, language }
       }
 
       yield put(push)
+    } else if (path === `/${city}/${language}`) {
```

Comment:
Why are you checking here for the root category? Shouldn't the PushCategoryLanguagesActionType also be put for other categories?

Reply:
Nope. If the fetch failed for some other reason, we throw an error right? `languageValid` is only `false` if the language not exists (for the root category). But if the language does not exist for the root category, it surely does not exist for any other category. Or am I missing something here?

Reply:
Nope, I don't think so. `loadCityContent` only returns false if the language is not available for the root category, otherwise it should throw. But if the language  is not available for the root category, it is surely not available in any other category. Or am I missing something here?

Reply:
Same should apply to events. If the language is not available for the events list, it is also not available for a single event

Reply:
True. We definitely know, that the language is not available for a specific page, if `loadCityContent` reutrns false. But in that case we are only given the information `load /augsburg/invalid-language/some-page` where we cannot determine any available language. So the only sensible thing we can do, is to show an error, that the requested page is not available at all.

Reply:
Makes sense, yes :)